tron-common/README.md

```diff
@@ -0,0 +1,42 @@
+# TRON Foundational Modules
+
+The **TRON Foundational Modules** Substreams contains a set of modules that allow you to easily retrieve basic information from the TRON blockchain.
+
+## Modules
+
+### map_transactions
+
+This module retrieves all the **NOT** failed transactions, without any more filtering.
+
+### filtered_transactions
+
+This module uses the `index_transactions` cache to match the filtered transactions based on the parameters passed as input to the module.
+
+You can directly _use_ this module to retrieve filtered transactions:
+
+```yaml
+modules:
+    - name: my_module
+      use: tron_common:filtered_transactions
+
+params:
+    my_module: source_account:account1
+```
+
+### index_transactions
+
+This module creates a cache of transactions based on:
+- The _contract type_ of every transactions.
+- The _owner address_ of every transactions that support it.
+- The _to address_ of every transactions that suppoirt it
+
+You can use this module as a `blockFilter` to filter transactions based on the parameters specified above:
+
+```yaml
+  - name: my_module
+    ...
+    blockFilter:
+      module: index_transactions
+      query:
+        string: (contract_type:contract_type1 || from:tron_address || to:tron_address)
+```
```

Comment:
Can you add a real working query here, like maybe something around USDT with both a `contract_type` and `to: <usdt>` something like

Reply:
hmm, I believe what you'd want here is an example of say, USDT transfers to a specific address right?

In the current setup, that's not possible because we don't index the contract address. If I were to do like `contract_type:TransferAssetContract && to:TR7NHqjeKQxGTCi8q8ZY4pL8otSzgjLj6t` where `TR7NHqjeKQxGTCi8q8ZY4pL8otSzgjLj6t` is USDT address, we'd only get the transfers that were made to USDT, which I don't think is what we want.

Do you think we'd want to add `contract_address` as a key to filters like `contract_address: TR7NHqjeKQxGTCi8q8ZY4pL8otSzgjLj6t && to:tron_address` ?

Reply:
Let's add for all contract address we find like the others.